.github/workflows/ci.yml

```diff
@@ -13,37 +13,40 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        cabal: ["3.2"]
+        cabal: ["3.4"]
         ghc:
-          - "8.10.1"
-          - "8.10.2"
+          - "9.0.0.20201227"
 
     steps:
     - uses: actions/checkout@v2
-      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/ghc-8.10'
+      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/ghc-9.0'
```

Comment:
What would go wrong without this line (i.e. if it is removed entirely)? cc @chshersh

Reply:
Nothing should go wrong. This is just an optimization. According to docs, possible events that trigger GitHub Actions include a lot of things, and by default CI is running a lot of times for different type events:

* https://docs.github.com/en/developers/webhooks-and-events/github-event-types

My understanding of this, is this is configured that way mostly because GitHub Actions can do a lot of things, not just build the code, but also review, add labels, write messages, etc. However, I usually only want to build on code changes for pull requests and commits to the main branches, and don't want to spend GitHub Actions resources on other events.

Reply:
OK, but doesn’t this `if` only apply to the checkout step? The other steps would run but without cloning the sources, and then fail

Reply:
Good point. Feel free to remove this line. I guess, nothing should become broken, and the config can be simplified.

I reread the documentation again, and it seems that the correct way to filter by events is something like this:

* https://github.com/Exheredrey/Exheredrey/blob/5a11c102af246b144ced9674dfb28ef235fd8d9f/.github/workflows/clang.yml#L5-L7